src/cmd/services/m3aggregator/config/server.go

```diff
@@ -24,15 +24,39 @@ import (
 	"time"
 
 	"github.com/m3db/m3/src/aggregator/server/http"
+	"github.com/m3db/m3/src/aggregator/server/m3msg"
 	"github.com/m3db/m3/src/aggregator/server/rawtcp"
 	"github.com/m3db/m3/src/metrics/encoding/msgpack"
 	"github.com/m3db/m3/src/metrics/encoding/protobuf"
+	"github.com/m3db/m3/src/msg/consumer"
 	"github.com/m3db/m3/src/x/instrument"
 	"github.com/m3db/m3/src/x/pool"
 	"github.com/m3db/m3/src/x/retry"
 	xserver "github.com/m3db/m3/src/x/server"
 )
 
+// M3MsgServerConfiguration contains M3Msg server configuration.
+type M3MsgServerConfiguration struct {
+	// Server is the server configuration.
+	Server xserver.Configuration `yaml:"server"`
```

Comment:
Should these be nil-able for defaults?

Reply:
I prefer to keep the non-nilable and then make `M3MsgServerConfiguration` itself at top level nil-able. 

Works well (i.e. not enforced unless you're opting into using m3msg server, and then it's forced to not be nil).